src/entities.rs

```diff
@@ -25,22 +27,26 @@ impl Entity {
     ///
     /// No particular structure is guaranteed for the returned bits.
     ///
+    /// If bits does not correspond to a possible `NonZeroU64` value returns `None`
+    ///
     /// Useful for storing entity IDs externally, or in conjunction with `Entity::from_bits` and
     /// `World::spawn_at` for easy serialization. Alternatively, consider `id` for more compact
     /// representation.
-    pub fn to_bits(self) -> u64 {
-        u64::from(self.generation) << 32 | u64::from(self.id)
+    pub fn to_bits(self) -> Option<NonZeroU64> {
+        NonZeroU64::new(NonZeroU64::from(self.generation).get() << 32 | u64::from(self.id))
```

Comment:
This should not return `Option`, because every `Entity` is guaranteed to have a (nonzero) bitpattern. Additionally, there's no point using `NonZeroU64::from` internally only to immediately convert back to a regular `u64`.

```suggestion
    pub fn to_bits(self) -> NonZeroU64 {
        unsafe { NonZeroU64::new_unchecked(u64::from(self.generation.get()) << 32 | u64::from(self.id)) }
```